pandas/core/generic.py

```diff
@@ -8050,6 +8041,17 @@ def interpolate(
                 downcast=downcast,
             )
         else:
+            obj, should_transpose = (self.T, True) if axis == 1 else (self, False)
+            if np.any(obj.dtypes == object):
```

Comment:
this is nitpicky, but doing this after the self.empty check means we wont warn in some cases

Reply:
Fixed